core/options.cpp

```diff
@@ -104,6 +104,33 @@ static int xioctl(int fd, unsigned long ctl, void *arg)
 	return ret;
 }
 
+static bool set_subdev_hdr_ctrl(int en)
+{
+	bool changed = false;
+	// Currently this does not exist in libcamera, so go directly to V4L2
+	// XXX it's not obvious which v4l2-subdev to use for which camera!
+	for (int i = 0; i < 8; i++)
+	{
+		std::string dev("/dev/v4l-subdev");
+		dev += (char)('0' + i);
+		int fd = open(dev.c_str(), O_RDWR, 0);
+		if (fd < 0)
+			continue;
+
+		v4l2_control ctrl { V4L2_CID_WIDE_DYNAMIC_RANGE, en };
+		xioctl(fd, VIDIOC_G_CTRL, &ctrl);
+		if (ctrl.value != en)
+		{
+			ctrl.value = en;
+			xioctl(fd, VIDIOC_S_CTRL, &ctrl);
+			changed = true;
+		}
+		close(fd);
+	}
```

Comment:
This will work but it's a bit hard to understand.

For a non-IMX708, the G_CTRL will fail (but that's OK as we don't set the changed flag and will try other devices).
For an IMX708 that's already in use, the S_CTRL may fail (changed may be set pessimistically but it will do no harm).

Is it helpful to test the return values of the ioctls? If not an explanatory comment would be good.

Reply:
Good point, I'll make a change to test the ioctl return value.